### DISCUSSION TITLE
Is reverse playback in model animations supported?

### DISCUSSION THREAD

**Discussion by ninjz** [sceneview/sceneform-android#255] [Q&A] [answered]
I was going to go for the Sceneform@1.15.0 library but saw that reverse playback is not supported on model animations. Is this still a limitation in this library?

Sort of a deal breaker for me, but I'd be open to any suggestions of alternative lightweight libs which support this. Thanks!

**Comment by ThomasGorisse** [comment]
What do you mean by revrse playback exactly? Reading the animation from end to begin?

**Reply by ninjz** [reply]
Yes exactly, playing the animation from end to beginning. Usually that would mean say setting the speed value to -1. The documentation from Sceneform 1.15.0 says specifically:
> "Reverse playback is not supported in model animations"

[reference from docs](https://developers.google.com/sceneform/develop/animation#sceneform_model_animation_vs_android_property_animation)

I do this fine on iOS using SceneKit, but see that this isn't supported here. My animations need to play in reverse and I don't want to need to have a separate animation just to play it in reverse which would bloat the application size.

**Reply by ThomasGorisse** [reply]
Model Animations on Sceneview/Sceneform are standard Android ObjectAnimator which are basely mapped to your animations frames.
You can do everything you do with the [ObjectAnimator](https://developer.android.com/reference/android/animation/ObjectAnimator)

```kotlin
val animation = modelNode.renderableInstance.animate(repeat)
animation.reverse()
```
>Edit: ~~`animation.start()`~~

See https://github.com/SceneView/sceneform-android#animations

**Reply by ninjz** [reply]
Sorry for the delay @ThomasGorisse . I just gave this a try and `.reverse()` seems to have no effect and the animation still plays in the same direction. 

Any other ideas?

**Comment by ninjz** [comment, marked as answer]
After diving a bit deeper into the ValueAnimator code, it works by just calling `.reverse()` as the call starts playing the animation in reverse and `.start()` does not need to be called after as it will result in playing the animation in the forward direction again. Thanks for your help @ThomasGorisse 

```kt
val animation = modelNode.renderableInstance.animate(repeat)
animation.reverse()
```
